test/com/lynxanalytics/biggraph/frontend_operations/LogisticRegressionTest.scala

```diff
@@ -11,7 +11,8 @@ class LogisticRegressionTest extends OperationsTestBase {
         Map("output" -> "label", "expr" -> "if (age > 30) 1.0 else 0.0"))
       .box(
         "Train a logistic regression model",
-        Map("name" -> "model", "label" -> "label", "features" -> "age", "max_iter" -> "20"))
+        Map("name" -> "model", "label" -> "label", "features" -> "age", "max_iter" -> "5",
+          "elastic_net_param" -> "0.8", "reg_param" -> "0.3"))
```

Comment:
This small change hides a full days investigation into why it was failing. The `model.transform` output before this PR:

```
---+--------+-----+--------------------+--------------------+--------------+
| id|features|label|   rawClassification|         probability|classification|
+---+--------+-----+--------------------+--------------------+--------------+
|  0|  [20.3]|  0.0|[12.3020660666562...|[0.99999545767066...|           0.0|
|  1|  [18.2]|  0.0|[14.0812553077173...|[0.99999923336580...|           0.0|
|  2|  [50.3]|  1.0|[-13.114923091360...|[2.01493200753395...|           1.0|
|  3|   [2.0]|  0.0|[27.8064294530462...|[0.99999999999916...|           0.0|
+---+--------+-----+--------------------+--------------------+--------------+
```

The probabilities in a more readable way:
```
Map(0 -> 0.9999954576706691, 1 -> 0.9999992333658014, 2 -> 2.014932007576853E-6, 3 -> 0.9999999999991609)
Map(0 -> 4.542329330892159E-6, 1 -> 7.666341985679281E-7, 2 -> 0.9999979850679924, 3 -> 8.391065620116933E-13)
```

After the PR but before this patch:
```
+---+--------+-----+--------------------+--------------------+--------------+
| id|features|label|   rawClassification|         probability|classification|
+---+--------+-----+--------------------+--------------------+--------------+
|  0|  [20.3]|  0.0|[12.4352283471900...|[0.99999602399226...|           0.0|
|  1|  [18.2]|  0.0|[14.2112159654392...|[0.99999932679546...|           0.0|
|  2|  [50.3]|  1.0|[-12.936023342083...|[2.40965794578568...|           0.0|
|  3|   [2.0]|  0.0|[27.9116918776468...|[0.99999999999924...|           0.0|
+---+--------+-----+--------------------+--------------------+--------------+
Map(0 -> 0.9999960239922627, 1 -> 0.9999993267954652, 2 -> 2.409657945785681E-6, 3 -> 0.9999999999992448)
Map(0 -> 3.976007737271914E-6, 1 -> 6.732045347845883E-7, 2 -> 0.9999975903420543, 3 -> 7.551737013500315E-13)
```

So the probabilities are fine, but the classification became all zero! Maybe the threshold is wrong? It's based on the F1 curve, which looks like this:
(before)
```
+--------------------+------------------+
|           threshold|         F-Measure|
+--------------------+------------------+
|  0.9999979850679924|               1.0|
|4.542329330943592E-6|0.6666666666666666|
|7.666341985190082E-7|               0.5|
|8.391143092838655...|               0.4|
+--------------------+------------------+
```
(after)
```
+--------------------+------------------+
|           threshold|         F-Measure|
+--------------------+------------------+
|  0.9999975903420543|               1.0|
|3.976007737271914E-6|0.6666666666666666|
|6.732045347845883E-7|               0.5|
|7.551737013500315...|               0.4|
+--------------------+------------------+
```

In both cases the highest F1 score (1) corresponds to the first line, so that threshold value is used then. You can see that these thresholds exactly match the class 1 probability for old Bob. The threshold is not applied as a simple comparison. It's used to find the maximum probability in https://github.com/apache/spark/blob/v3.1.1/mllib/src/main/scala/org/apache/spark/ml/classification/ProbabilisticClassifier.scala#L209. This code has not changed between Spark 2.4.3 and Spark 3.1.1.

My theory is that the comparison was exactly on the knife's edge, and due to bad luck it now falls in the wrong direction.

I hope in a better dataset (more than a single class 1 sample) the threshold would be better placed and this issue would not come up. To get the test to pass I added and set the elastic net parameters that all the Spark examples use. Let me know if you can think of a better solution.